.github/workflows/continuous-integration-workflow.yml

```diff
@@ -0,0 +1,310 @@
+      - name: Install Yarn dependencies
+        run: make yarndep
+      - name: Cache Go vendor packages
+        uses: actions/cache@v1
+        with:
+          path: /go/pkg/mod
+          key: $${{ secrets.CACHE_VERSION }}-go-mod-$${ hashFiles('go.sum') }}
+          restore-keys: |
+            $${{ secrets.CACHE_VERSION }}-go-mod
+      - name: Download Go vendor packages
+        run: go mod download
+      - name: Set up Yarn for Contracts
+        run: yarn setup:contracts
+      - name: Run tests
+        run: go test -v -p 6 -parallel 4 ./...
+        env:
+          DATABASE_URL: postgres://postgres:postgres@postgres:5432/postgres?sslmode=disable
+
+  rust:
+    name: Rust
+    runs-on: ubuntu-latest
+    container:
+      image: smartcontract/builder:1.0.31
+      env:
+        CARGO_HOME: /root/.cargo
+        RUSTUP_HOME: /root/.rustup
+    steps:
+      - name: Checkout the repo
+        uses: actions/checkout@v2
+      - name: Run Rust test
+        run: ./tools/ci/rust_test
+
+  geth-postgres:
+    name: Geth Postgres
+    runs-on: self-hosted
+    env:
+      GETH_MODE: true
+      CI: true
+    steps:
+      - name: Checkout the repo
+        uses: actions/checkout@v2
+      - name: Run tests
+        run: cd ./tools/docker && timeout --foreground 1200s ./compose test
+
+  parity-postgres:
+    name: Parity Postgres
+    runs-on: self-hosted
+    env:
+      CI: true
+    steps:
+      - name: Checkout the repo
+        uses: actions/checkout@v2
+      - name: Run tests
+        run: cd ./tools/docker && timeout --foreground 1200s ./compose test
+
+  solidity:
+    name: Solidity
+    runs-on: ubuntu-latest
+    container: smartcontract/builder:1.0.31
+    steps:
+      - name: Checkout the repo
+        uses: actions/checkout@v2
+      - name: Cache Yarn dependencies
+        uses: actions/cache@v1
+        with:
+          path: /usr/local/share/.cache/yarn
+          key: $${{ secrets.CACHE_VERSION}}-yarn-$${ hashFiles('yarn.lock') }}
+          restore-keys: |
+            $${{ secrets.CACHE_VERSION}}-yarn
+      - name: Install Yarn dependencies
+        run: make yarndep
+      - name: Cache Python dependencies
+        uses: actions/cache@v1
+        with:
+          path: ~/.cache/pip
+          key: $${{ secrets.CACHE_VERSION}}-pip-$${ hashFiles('requirements.txt') }}
+          restore-keys: |
+            $${{ secrets.CACHE_VERSION}}-pip
+      - name: Install python packages
+        run: pip3 install -r requirements.txt
+      - name: Run tests
+        run: ./tools/ci/solidity_test
+
+  styleguide:
+    name: Styleguide
+    runs-on: ubuntu-latest
+    container: smartcontract/builder:1.0.31
+    steps:
+      - name: Checkout the repo
+        uses: actions/checkout@v2
+      - name: Cache Yarn dependencies
+        uses: actions/cache@v1
+        with:
+          path: /usr/local/share/.cache/yarn
+          key: $${{ secrets.CACHE_VERSION}}-yarn-$${ hashFiles('yarn.lock') }}
```

Comment:
I think we should make the package manager substring be the prefix, that way if we ever have a restore key of `yarn-`, it would still be valid, but a restore key of `${{secrets.CACHE_VERSION}}` would mangle our cache with other cached dependencies

Reply:
I believe based on the way [restore-key lookup](https://help.github.com/en/actions/configuring-and-managing-workflows/caching-dependencies-to-speed-up-workflows#matching-a-cache-key) works that if we don't but the CACHE_VERSION first, we can't be sure we're invalidating the cache by changing the key. If the restore keys were `yarn-` and `yarn-<cache_version>` if you change the cache version it'll still restore from some other matching cache based on the `yarn-` restore key which means you potentially wouldn't be able to start with fresh dependencies.

Reply:
Oh yeah, for sure. We would need to have `yarn-<cache_version>` as the smallest string, but I'm saying that we should make the suggest switch just in case a change is made in the future, having `yarn-` rather than `<cache_version>-` as the minimal restore key would be much safer.